index.html

```diff
@@ -1320,63 +1320,200 @@ <h1>Core Properties</h1>
+            </tr>
+          </tbody>
+        </table>
+      </section>
+
+      <section>
+        <h2>Core Properties for a <a>Verification Method</a></h2>
+
+        <table class=simple>
+          <thead>
+            <tr>
+              <th>Property</th>
+              <th>Usage constraints</th>
+              <th>Value constraints</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>id</code></td>
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URI</a>-s and which SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>controller</a></code></td>
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> or an <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a> that conform to the rules in Section <a href="#did-syntax"></a>.
+              </td> 
+            </tr>
+            <tr>
+              <td><code>type</code></td>
+              <td>REQUIRED</td>
+              <td>
+                A <a data-cite="INFRA#string">string</a> or an <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code><a>publicKeyJwk</a></code></td>
+              <td>OPTIONAL</td>
+              <td>A <a data-cite="INFRA#maps">map</a> representing a JSON Web Key that conforms to [[RFC7517]].</td>
+            </tr>
+            <tr>
+              <td><code><a>publicKeyBase58</a></code></td>
+              <td>OPTIONAL</td>
+              <td>A <a
+                data-cite="INFRA#string">string</a> that conforms to a base58btc encoded public key.</td>
+            </tr>
+          </tbody>
+        </table>
+
+        <p>As an extra constraint on the <code><a>publicKeyJwk</a></code> and <code><a>publicKeyBase58</a></code> properties, while both of these properties are OPTIONAL, a particular <a>Verification Method</a> MUST contain <em>exactly one</em> of the two.</p>
+      </section>
+
+
+      <section>
+        <h2>Core Properties for a <a>Service Endpoint</a></h2>
+
+      <table class=simple>
+          <thead>
+            <tr>
+              <th>Property</th>
+              <th>Usage constraints</th>
+              <th>Value constraints</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>id</code></td>
+              <td>REQUIRED</td>
+              <td>
+                when used on a <a>Service Endpoint</a>: a <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URI</a>-s.
```

Comment:
```suggestion
                a <a data-cite="INFRA#string">string</a> that conforms to the rules of [[RFC3986]] for <a>URIs</a>.
```

I think the "when used on a service endpoint" part is implicit by the table